src/query/sql/src/planner/optimizer/s_expr.rs

```diff
@@ -171,9 +171,7 @@ impl SExpr {
     /// Check if contain subquery
     pub(crate) fn contain_subquery(&self) -> bool {
         if !find_subquery(&self.plan) {
-            for child in self.children.iter() {
-                return child.contain_subquery();
-            }
+            return self.children.iter().any(|child| child.contain_subquery());
```

Comment:
should check if children are empty, it will return true.

Reply:
!self.children.is_empty() && ...

Reply:
An empty iterator returns false

<img width="1016" alt="image" src="https://user-images.githubusercontent.com/41979257/221159540-c8f9ff95-9685-40f1-aa7b-bdc09fa1d060.png">